### PR TITLE
update library to make apollo functionv4 and apollo server work together

### PR DESCRIPTION
**Pull request by nguyenduy**
## Summary by Sourcery

Update the Azure Cognitive Search implementation to improve performance by caching search clients.

New Features:
- Add an initialization step for search clients.

Enhancements:
- Cache search clients to improve search performance.

Tests:
- Update tests to reflect changes in the search service.

### CODE REVIEW

File: data-access/seedwork/services-seedwork-cognitive-search-az/index.ts [hunk 1]
```diff
@@ -59,23 +64,37 @@ export class AzCognitiveSearch implements CognitiveSearchBase {
...
-    let searchClient = this.client.getSearchClient(indexName);
-    const result = await searchClient.search(searchText, options);
-    console.log("search result", result);
-    return result;
+    const startTime = new Date();
+    const result = await this.searchClients.get(indexName).search(searchText, options);
+     const endTime = new Date();
+     console.log(`SearchLibrary took ${endTime.getTime() - startTime.getTime()}ms`);
+     return result
   }
```
[the remark this answers is not in the document]
Comment (nguyenduy) [reply data-access/seedwork/services-seedwork-cognitive-search-az/index.ts:72]: fixed

File: data-access/seedwork/services-seedwork-cognitive-search-az/index.ts [hunk 2]
```diff
@@ -25,30 +26,34 @@ export class AzCognitiveSearch implements CognitiveSearchBase {
     this.client = new SearchIndexClient(endpoint, credentials);
   }
 
+  async initializeSearchClients(): Promise<void> {
+    const indexNames = this.client.listIndexesNames();
+    for await (const indexName of indexNames) {
+      this.searchClients.set(indexName, this.client.getSearchClient(indexName));
+    }
+  }
```
[the remark this answers is not in the document]
Comment (nguyenduy) [reply data-access/seedwork/services-seedwork-cognitive-search-az/index.ts:34]: thanks

File: data-access/seedwork/services-seedwork-cognitive-search-az/index.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { CognitiveSearchBase } from "../services-seedwork-cognitive-search-inter
 
 export class AzCognitiveSearch implements CognitiveSearchBase {
```
[the remark this answers is not in the document]
Comment (nguyenduy) [reply data-access/seedwork/services-seedwork-cognitive-search-az/index.ts:5]: great idea

**Comment by nguyenduy** [comment]
@gidich it seems that I accidentally added you as a reviewer. I requested Nick to review this PR already, so I am removing you.

**Review by nnoce14 — APPROVED** [review]
Looks good, thanks Duy!

**Pull request merged**